test/fpm_test/test_filesystem.f90

```diff
@@ -96,11 +98,102 @@ subroutine check_string(error, actual, expected)
 
         if (actual /= expected) then
             call test_failed(error, &
-                "Character value missmatch "//&
+                "Character value mismatch "//&
                 "expected '"//expected//"' but got '"//actual//"'")
         end if
 
     end subroutine check_string
 
 
+    subroutine test_mkdir_rmdir(error)
+
+        !> Error handling
+        type(error_t), allocatable, intent(out) :: error
+
+        logical :: is_win
+
+        is_win = (get_os_type() == OS_WINDOWS)
+
+        if (is_win) then
+            call check_mkdir(error, "tmpdir\subdir")
+          else
+            call check_mkdir(error, "tmpdir/subdir")
+        end if
+        if (allocated(error)) return
+
+        if (is_win) then
+            call check_rmdir(error, "tmpdir\subdir")
+          else
+            call check_rmdir(error, "tmpdir/subdir")
+        end if
+        if (allocated(error)) return
+
+        call check_rmdir(error, "tmpdir")
+        if (allocated(error)) return
+
+    end subroutine test_mkdir_rmdir
+
+
+    !> Create a directory and verify its existence
+    subroutine check_mkdir(error, path)
+
+        !> Error handling
+        type(error_t), allocatable, intent(out) :: error
+
+        !> Directory path
+        character(len=*), intent(in) :: path
+
+        logical :: stat
+
+        ! Directory shouldn't exist before it's created
+        stat = (is_dir(path) .eqv. .false.)
+        if (.not. stat) then
```

Comment:
If `stat` isn't required elsewhere, then we can be more concise with these checks to avoid confusion:
```suggestion
        ! Directory shouldn't exist before it's created
        if (is_dir(path)) then
```